### PR TITLE
[#3535] Send messages to Whatsapp Cloud

### LINKED ISSUE #3535: Whatsapp cloud source: Send messages
[Parent issue](https://github.com/airyhq/airy/issues/3464)

We should be able to send messages to the source (https://developers.facebook.com/docs/whatsapp/cloud-api/guides/send-messages)

Should also support template messages: https://developers.facebook.com/docs/whatsapp/cloud-api/guides/send-message-templates

### PR DESCRIPTION
**Pull request by chrismatix**
resolves #3535

### CODE REVIEW

**Comment by chrismatix** [comment]
It works!

File: backend/api/websocket/src/test/java/co/airy/core/api/websocket/WebSocketControllerTest.java [hunk 1]
```diff
@@ -184,7 +184,7 @@ void canSendTagEvents() throws Exception {
     private static StompSession connectToWs(int port) throws ExecutionException, InterruptedException {
         final WebSocketStompClient stompClient = new WebSocketStompClient(new StandardWebSocketClient());
         MappingJackson2MessageConverter messageConverter = new MappingJackson2MessageConverter();
-        ObjectMapper objectMapper = new ObjectMapper().setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);
```
Comment (chrismatix) [inline backend/api/websocket/src/test/java/co/airy/core/api/websocket/WebSocketControllerTest.java:187]: Also refactored this deprecated import

File: backend/sources/facebook/connector/src/main/java/co/airy/core/sources/facebook/Connector.java [hunk 2]
```diff
@@ -88,6 +88,7 @@ public List<KeyValue<String, SpecificRecordBase>> sendMessage(SendMessageRequest
                 results.add(KeyValue.pair(getId(errorPayload).toString(), errorPayload));
             }
             updateDeliveryState(message, DeliveryState.FAILED);
+            results.add(KeyValue.pair(message.getId(), message));
```
Comment (chrismatix) [inline backend/sources/facebook/connector/src/main/java/co/airy/core/sources/facebook/Connector.java:91]: This fixes a sofar unknown bug

File: backend/sources/whatsapp/connector/src/main/java/co/airy/core/sources/whatsapp/api/MetaConfig.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class MetaConfig {
+    @Bean
+    @Qualifier("metaObjectMapper")
```
Comment (chrismatix) [inline backend/sources/whatsapp/connector/src/main/java/co/airy/core/sources/whatsapp/api/MetaConfig.java:13]: So far the meta-object mapper has been the same as ours in terms of config. It's better however to explicitly split them.

File: backend/sources/whatsapp/connector/src/test/java/co/airy/core/sources/whatsapp/SendMessageTest.java [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+                                .setSenderId(sourceConversationId)
+                                .setDeliveryState(DeliveryState.DELIVERED)
+                                .setConversationId(conversationId)
+                                .setChannelId(channelId)
+                                .setContent("{\"text\":\"" + text + "\"}")
+                                .setIsFromContact(true)
+                                .build())
+        ));
+
+        TimeUnit.SECONDS.sleep(5);
```
Comment (juan-sebastian) [inline backend/sources/whatsapp/connector/src/test/java/co/airy/core/sources/whatsapp/SendMessageTest.java:138]: with the following attributes
```
kafka.commit-interval-ms=0
kafka.cache.max.bytes=0
```

You should not need to wait that long, 100ms will be enough? Also there is other way to get around the sleep time?

Comment (chrismatix) [reply backend/sources/whatsapp/connector/src/test/java/co/airy/core/sources/whatsapp/SendMessageTest.java:137]: Just tested this and it doesn't make a difference. I think that this time is genuinely spent on catching up with the streams app.

**Review by juan-sebastian — APPROVED** [review]

**Pull request merged**